src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -67,6 +73,39 @@ public object Execute(object target, object[] parameters)
             return _executor(target, parameters);
         }
 
+        public object GetDefaultValueForParameter(int index)
+        {
+            Debug.Assert(ActionParameters.Length > 0);
```

Comment:
This is a `public` method and should not contain debug assertions related to properties initialized in the constructor. Instead `throw` an `ArgumentOutOfRangeException` if `index` is less than `0` or greater than `ActionParameters.Length - 1`.